src/decrypt.rs

```diff
@@ -55,35 +56,43 @@ pub async fn try_decrypt(
     .await
 }
 
-pub async fn create_decryption_info(
+pub async fn prepare_decryption(
     context: &Context,
     mail: &ParsedMail<'_>,
+    from: &[SingleInfo],
     message_time: i64,
 ) -> Result<DecryptionInfo> {
-    let from = mail
-        .headers
-        .get_header(HeaderDef::From_)
-        .and_then(|from_addr| mailparse::addrparse_header(from_addr).ok())
-        .and_then(|from| from.extract_single_info())
-        .map(|from| from.addr)
-        .unwrap_or_default();
-
-    let autocrypt_header = Aheader::from_headers(&from, &mail.headers)
+    let from = if let Some(f) = from.first() {
+        &f.addr
+    } else {
+        return Ok(DecryptionInfo::default());
```

Comment:
Yes, probably best to postpone this to the next PR - I guess if we want to do this, it will involve letting `from_bytes_with_partial()` return `std::Result<Self, MimeparserError>`, where MimeparserError is an enum {sql, CompletelyWrongSyntax, SomeWrongSyntax { rfc724_mid: String }}

Note that currently (and independently of this PR) we do add all the information we do have about the message to the database entry, and it's not even trashed. Which does have advantages, because, showing a message in a weird way is usually better than not showing it and silently ignoring it. Would be interesting how these messages without From are shown in the UI, maybe I'll try this out. Trying to log them in `test_no_from()`:
```
========== Chats of alice: ==========
Group#Chat#10: foo [3 member(s)] 
--------------------------------------------------------------------------------
Can't log message: invalid contact: Query returned no rows
--------------------------------------------------------------------------------

Group#Chat#10: foo [3 member(s)] 
--------------------------------------------------------------------------------
Can't log message: invalid contact: Query returned no rows
--------------------------------------------------------------------------------
```

Reply:
I did try it out, but it isn't handled nicely at all currently. When clicking on the "foo" chat, nothing happens:

![Uploading Screenshot_2022-10-18_21-45-03.jpg…]()